tokio/tests/io_fill_buf.rs

```diff
@@ -0,0 +1,34 @@
+#![warn(rust_2018_idioms)]
+#![cfg(feature = "full")]
+
+use tempfile::NamedTempFile;
+use tokio::fs::File;
+use tokio::io::{AsyncBufReadExt, BufReader};
+use tokio_test::assert_ok;
+
+#[tokio::test]
+async fn fill_buf_file() {
```

Comment:
take it or leave it: maybe worth a comment summarizing the bug this is a regression test for (and a link to the original issue?)

Reply:
I don't really feel like waiting for CI over again, since I am making a release immediately after merging this.